awswrangler/distributed/ray/_register.py

```diff
@@ -0,0 +1,53 @@
+"""Ray and Modin registered methods (PRIVATE)."""
+# pylint: disable=import-outside-toplevel
+from awswrangler._distributed import MemoryFormatEnum, engine, memory_format
+from awswrangler._utils import table_refs_to_df
+from awswrangler.distributed.ray._core import ray_remote
+from awswrangler.lakeformation._read import _get_work_unit_results
+from awswrangler.s3._delete import _delete_objects
+from awswrangler.s3._read_parquet import _read_parquet, _read_parquet_metadata_file
+from awswrangler.s3._read_text import _read_text
+from awswrangler.s3._select import _select_object_content, _select_query
+from awswrangler.s3._wait import _wait_object_batch
+from awswrangler.s3._write_dataset import _to_buckets, _to_partitions
+from awswrangler.s3._write_parquet import _to_parquet, to_parquet
+from awswrangler.s3._write_text import _to_text, to_csv, to_json
+
+
+def register_ray() -> None:
+    """Register dispatched Ray and Modin (on Ray) methods."""
+    # LakeFormation
+    engine.register_func(_get_work_unit_results, ray_remote(_get_work_unit_results))
+
+    # S3
+    engine.register_func(_delete_objects, ray_remote(_delete_objects))
+    engine.register_func(_read_parquet_metadata_file, ray_remote(_read_parquet_metadata_file))
+    engine.register_func(_select_query, ray_remote(_select_query))
+    engine.register_func(_select_object_content, ray_remote(_select_object_content))
+    engine.register_func(_wait_object_batch, ray_remote(_wait_object_batch))
```

Comment:
What do you think about keeping the register only for registration and moving the function definitions to modin module with the others? E.g.
```
distributed/ray/modin/s3/...py:

_wait_object_batch_distributed = ray_remote(_wait_object_batch)
_select_object_content_distributed = ray_remote(_select_object_content)
```
Also it seems now we do not actually need our own `ray_remote` decorator that handled both distributed and non-distributed calls dynamically and we can use default `@remote`.